src/components/Carousel/index.tsx

```diff
@@ -0,0 +1,343 @@
+  switch (breakpoint) {
+    case 'DESKTOP':
+      return `@media (max-width: ${BREAKE_POINTS.DESKTOP}px)`
+    case 'MOBILE':
+      return `@media (max-width: ${BREAKE_POINTS.MOBILE}px)`
+    case 'TABLET':
+      return `@media (max-width: ${BREAKE_POINTS.TABLET}px)`
+    default:
+      return ``
+  }
+}
+
+export interface CarouselProps {
+  images: { url: string; id: number }[]
+  isArrow: boolean
+  size?: number
+  name: string
+}
+
+interface SliderProps {
+  cursorOn: boolean
+  size: number
+}
+
+interface CurrentIndicatorProps {
+  imageIndex: number
+}
+
+interface ImageBoxProps {
+  currentImageValue: number
+}
+
+interface ImageProps {
+  size: number
+}
+
+interface ArrowProps {
+  currentImageValue: number
+}
+
+interface IndicatorBoxProps {
+  isArrow: boolean
+}
+
+type HandleOffset = (navType: keyof typeof NAV_TYPE) => void
+const NAV_TYPE = {
+  LEFT: 'LEFT',
+  RIGHT: 'RIGHT'
+} as const
+
+const ARROW_RIGHT = ICON.CHEVRON_RIGHT_40
+const ARROW_LEFT = ICON.CHEVRON_LEFT_40
+
+const Carousel = ({
+  images,
+  isArrow,
+  size = 687,
+  name
+}: CarouselProps): ReactElement => {
+  const desktop = useMediaQuery(`(min-width:1023px)`)
+  const carouselWidthSize = desktop ? size : 100
+  const lastImageValue = carouselWidthSize * (images.length - 1)
+  const [currentImageValue, setCurrentImageValue] = useState<number>(0)
+  const [startClientX, setStartClientX] = useState<number>(0)
+  const [endClientX, setEndClientX] = useState<number>(0)
+  const [cursorOn, setCursorOn] = useState<boolean>(false)
+
+  const handleIndicator = (imageIndex: number): void => {
+    setCurrentImageValue((imageIndex - 1) * carouselWidthSize)
+  }
+
+  const TRANSLATE_VALUE_OF_NAV_TYPE = {
+    LEFT: -carouselWidthSize,
+    RIGHT: carouselWidthSize
+  }
+
+  const handleOffset: HandleOffset = navType => {
+    const { LEFT } = NAV_TYPE
+    const endPoint = navType === LEFT ? 0 : lastImageValue
+    const prevTranslateValue = navType === LEFT ? lastImageValue : 0
+    if (currentImageValue === endPoint) {
+      setCurrentImageValue(prevTranslateValue)
+    } else {
+      setCurrentImageValue(prev => prev + TRANSLATE_VALUE_OF_NAV_TYPE[navType])
+    }
+  }
+
+  const handleTouchStart: TouchEventHandler<HTMLDivElement> = e => {
+    setEndClientX(e.touches[0].clientX)
+    setCursorOn(true)
+  }
+  const handleTouchStartEnd: TouchEventHandler<HTMLDivElement> = e => {
+    setStartClientX(e.changedTouches[0].clientX)
+    setCursorOn(false)
+  }
+
+  useEffect(() => {
+    const userDragLength = 100
+    const dragSpace = Math.abs(startClientX - endClientX)
+    const userSlideRight =
+      endClientX > startClientX && dragSpace > userDragLength
+    const userSlideLeft =
+      endClientX < startClientX && dragSpace > userDragLength
+    if (startClientX === 0) {
+      return
+    }
+    if (userSlideRight) {
+      handleOffset(NAV_TYPE.RIGHT)
+    } else if (userSlideLeft) {
+      handleOffset(NAV_TYPE.LEFT)
+    }
+  }, [endClientX])
```

Comment:
```suggestion
  useEffect(() => {
    const dragSpace = Math.abs(startClientX - endClientX)
    const userSlideRight =
      endClientX > startClientX && dragSpace > USER_DRAG_LENGTH
    const userSlideLeft =
      endClientX < startClientX && dragSpace > USER_DRAG_LENGTH
      
    if (startClientX === 0) {
      return
    }
    
    if (userSlideRight) {
      handleOffset(NAV_TYPE.RIGHT)
    } else if (userSlideLeft) {
      handleOffset(NAV_TYPE.LEFT)
    }
  }, [endClientX])
```
상단에 선언된 `USER_DRAG_LENGTH` 활용하면 좋을 것 같아요!
개인적으로 적절한 공간은 코드 파악하는 데에 도움이 되는 것 같아요! ㅎㅎ